src/app/Dashboard/LayoutTemplatePicker.tsx

```diff
@@ -0,0 +1,578 @@
+    return (
+      <DeleteWarningModal
+        warningType={DeleteOrDisableWarningType.DeleteLayoutTemplate}
+        visible={isDeleteWarningModalOpen}
+        onClose={handleDeleteWarningModalClose}
+        onAccept={() => handleTemplateDelete(toDelete)}
+      />
+    );
+  }, [isDeleteWarningModalOpen, toDelete, handleDeleteWarningModalClose, handleTemplateDelete]);
+
+  const handleUploadButton = React.useCallback(() => {
+    setIsUploadModalOpen(true);
+  }, [setIsUploadModalOpen]);
+
+  const uploadButton = React.useMemo(
+    () => (
+      <Button
+        key="upload"
+        variant="secondary"
+        aria-label={t('DashboardLayoutToolbar.UPLOAD.LABEL')}
+        onClick={handleUploadButton}
+        icon={<UploadIcon />}
+        data-quickstart-id="dashboard-upload-btn"
+      >
+        {t('UPLOAD', { ns: 'common' })}
+      </Button>
+    ),
+    [t, handleUploadButton]
+  );
+
+  const onSearchChange = React.useCallback(
+    (value: string) => {
+      setSearchFilter(value);
+    },
+    [setSearchFilter]
+  );
+
+  const onDeleteChip = React.useCallback(
+    (_category: string, chip: string) => {
+      setSelectedFilters((prev) => prev.filter((item) => item !== chip));
+    },
+    [setSelectedFilters]
+  );
+
+  const onFilterSelect = React.useCallback(
+    (
+      _ev: React.MouseEvent | React.ChangeEvent,
+      selection: string | SelectOptionObject,
+      isPlaceholder: boolean | undefined
+    ) => {
+      const selected = selection as LayoutTemplateFilter;
+      setSelectedFilters((prev) => {
+        if (isPlaceholder) {
+          return [];
+        }
+        if (prev.includes(selected)) {
+          return prev.filter((item) => item !== selected);
+        }
+        return [...prev, selected];
+      });
+    },
+    [setSelectedFilters]
+  );
+
+  const onFilterSelectToggle = React.useCallback(
+    (isExpanded: boolean) => {
+      setIsFilterSelectOpen(isExpanded);
+    },
+    [setIsFilterSelectOpen]
+  );
+
+  const onClearAllFilters = React.useCallback(() => {
+    setSelectedFilters([]);
+  }, [setSelectedFilters]);
+
+  const onSortSelectToggle = React.useCallback(
+    (isExpanded: boolean) => {
+      setIsSortSelectOpen(isExpanded);
+    },
+    [setIsSortSelectOpen]
+  );
+
+  const onSortSelect = React.useCallback(
+    (
+      _ev: React.MouseEvent | React.ChangeEvent,
+      selection: string | SelectOptionObject,
+      isPlaceholder: boolean | undefined
+    ) => {
+      const selected = selection as LayoutTemplateSort;
+      setSelectedSort((_prev) => {
+        if (isPlaceholder) {
+          return undefined;
+        }
+        return selected;
+      });
+      setIsSortSelectOpen(false);
+    },
+    [setSelectedSort, setIsSortSelectOpen]
+  );
+
+  const sortArrowIcon = React.useMemo(() => {
+    if (!selectedSort) {
+      return <ArrowsAltVIcon />;
+    }
+    return sortDirection === 'asc' ? <LongArrowAltUpIcon /> : <LongArrowAltDownIcon />;
```

Comment:
Lets try this. Just more descriptive when used outside the table header.

![Screenshot from 2023-04-11 21-08-28](https://user-images.githubusercontent.com/68053619/231321306-82dd3e47-1fa8-4e5b-8e05-9b045f9e6b11.png)